src/layouts/sidebar-sidecar/utils/prepare-nav-data-for-client.ts

```diff
@@ -69,9 +70,16 @@ function prepareNavNodeForClient(node: NavNode, basePaths: string[]): MenuItem {
       id: slugify(`menu-item-${node.path}`, { lower: true }),
     }
   } else if (isNavDirectLink(node)) {
-    // For direct link nodes, add `id` only
+    // For direct links, if they're absolute URLs,
+    // such as https://www.example.com, then leave them untouched.
+    // Otherwise, prefix them with the product slug (basePath[0])
+    const adjustedHref = isAbsoluteUrl(node.href)
+      ? node.href
+      : `/${basePaths[0]}${node.href}`
+    // Also add `id` for direct links
     return {
       ...node,
+      href: adjustedHref,
```

Comment:
At present, folks sometimes use `NavDirectLink` nodes for _non-external_ URLs, such as `/api-docs`. Unlike absolute, external URLs, such as `https://www.example.com`, these URLs make assumptions about the URL structure of the `.io` sites, and these assumptions are not true in `dev-portal`, so we end up with broken links.

To fix this, if a `NavDirectLink` node has a non-absolute URL, we prefix it with the product.slug (which in this context is `basePath[0]`). For example, the [Consul docs sidebar](https://dev-portal-git-zsadd-consul-docs-views-hashicorp.vercel.app/consul/docs) has an API link with the URL `/api-docs`, and we prefix this to end up with the valid URL `/consul/api-docs`.

Reply:
I took a deeper look at this because I saw that external link icons were being rendered for the "API" and "Commands (CLI)" sidebar items on `/consul/docs` when that's not the desired behavior. To word what you already said in another way, sometimes `href` is used for specifying internal link paths instead of `path`, and this presents an issue in the `SidebarNavMenuItem` component that conditionally renders an external link icon based on the `fullPath` property that is only added to nav data if the `path` property is defined.

So here's how I approached handling that, I think there needs to be replacement of `href` with `path` and `fullPath` instead of always setting `href`, because the latter will tell the sidebar to render the external link:

```typescript
function prepareNavNodeForClient(node: NavNode, basePaths: string[]): MenuItem {
  if (isNavBranch(node)) {
    ...
  } else if (isNavLeaf(node)) {
    ...
  } else if (isNavDirectLink(node)) {
    const id = slugify(`external-url-${node.title}`, { lower: true })

    // check if there is data that disagrees with DevDot's assumptions/needs
    const hrefIsNotAbsolute = !isAbsoluteUrl(node.href)

    // if there is data that needs fixed, correct it by:
    // - removing the `href` prop
    // - adding the `path` prop
    // - adding the `fullPath` that is usually added in the previous block of this if-statement
    if (hrefIsNotAbsolute) {
      return {
        ...node,
        href: null,
        id,
        path: node.href,
        fullPath: `/${basePaths[0]}${node.href}`,
      }
    } else {
      // otherwise, just add the id prop as we did before
      return { ...node, id }
    }
  } else {
    ...
  }
}
```

Reply:
Ah thanks for this, this makes sense! Will take this approach 👍

One UX concern I had was that when these links are rendered without an "external" icon, it implies to me that'll they'll behave like any other sidebar link (ie the sidebar context stays consistent), so without the "external" icon it can feel jarring for the sidebar to shift. However, and I think this echoes your suggested approach, this is probably more a "tweak the sidebar content" issue, the "switch to `path` when internal" implementation seems more correct 👍💯

Reply:
I'd argue that the Consul nav data isn't currently being used correctly. The root nav items that link to API and Commands for example are redundant with the main navigation header, and I've noticed that some links actually link to other docs that live in a different section in the same sidebar. Both of these are misuses of the sidebar IMO, so I think we'll have to fix these in the future. But overall and what we can address in the short term: if they link to a page that lives in DevDot (or will live in DevDot very soon), then the external link icon shouldn't be rendered.

Reply:
+1 great points, fully agreed on this being a nav-data content issue. Have updated this with the suggested approach ☺️